ansible/roles/ansible-tvault-contego-extension/tasks/install_rhel.yml

```diff
@@ -15,36 +14,53 @@
 
 - name: Reload Daemon
   shell: systemctl daemon-reload
-
-- name: Generate modules.dep and map files
-  shell: depmod -a
-  register: depmod_reg
+- name: Run cmd to create nbd 128 devices
+  shell: sudo modprobe nbd nbds_max=128
+  register: cmd_nbd
   ignore_errors: yes
 
 - debug:
-    msg: "{{ depmod_reg }}"
-  when: depmod_reg.rc != 0
+    msg: "{{ cmd_nbd }}"
+  when: cmd_nbd.changed == False
+  ignore_errors: yes
 
-- name: Create nbd 128 devices
-  shell: sudo modprobe nbd nbds_max=128
-  register: crt_nbd
+- name: Create nbd config file
+  lineinfile:
+    path: /etc/modprobe.d/nbd.conf
+    line: 'options nbd nbds_max=128'
+    create: yes
+  ignore_errors: yes
+  
+- name: Load nbd module into Linux Kernel
+  lineinfile:
+    path: /etc/modules-load.d/openstack-ansible.conf
+    line: 'nbd'
+    create: no
   ignore_errors: yes
 
+- name: Run cmd to create rbd
+  shell: sudo modprobe rbd
+  register: cmd_rbd
+  ignore_errors: yes
+ 
 - debug:
-    msg: "{{ crt_nbd }}"
-  when: crt_nbd.rc != 0
+    msg: "{{ cmd_rbd }}"
+  when: cmd_rbd.changed == False
+  ignore_errors: yes
 
-- name: Load nbd 128 devices
+- name: Create rbd config file
   lineinfile:
-    path: /etc/rc.modules
-    line: 'depmod -a && sudo modprobe nbd nbds_max=128'
+    path: /etc/modprobe.d/rbd.conf
+    line: 'options rbd'
     create: yes
-  register: chk_nbd
   ignore_errors: yes
 
-- debug:
-    msg: "{{ chk_nbd }}"
-  when: crt_nbd.rc != 0
+- name: Load rbd module into Linux Kernel
+  lineinfile:
+    path: /etc/modules-load.d/openstack-ansible.conf
```

Comment:
Kolla-ansible code must be using template for this.
It will overwrite our changes.
Create new file for rbd.
rbd.conf

Reply:
In the above task, ansible is loading the rbd module into Linux kernel
It doesn't overwrite the changes cause we are using lininfile ansible module.